python/lsst/qa/explorer/functors.py

```diff
@@ -767,3 +793,169 @@ class Seeing(Functor):
     def _func(self, df):
         return 0.168*2.35*np.sqrt(0.5*(df['ext_shapeHSM_HsmPsfMoments_xx']**2 +
                                        df['ext_shapeHSM_HsmPsfMoments_yy']**2))
+
+
+class e1(Functor):
+    name = "Ellipticity e1"
+
+    def __init__(self, colXX, colXY, colYY, **kwargs):
+        self.colXX = colXX
+        self.colXY = colXY
+        self.colYY = colYY
+        self._columns = [self.colXX, self.colXY, self.colYY]
+        super().__init__(**kwargs)
+
+    @property
+    def columns(self):
+        return [self.colXX, self.colXY, self.colYY]
+
+    def _func(self, df):
+        return df[self.colXX] - df[self.colYY] / (df[self.colXX] + df[self.colYY])
+
+
+class e2(Functor):
+    name = "Ellipticity e2"
+
+    def __init__(self, colXX, colXY, colYY, **kwargs):
+        self.colXX = colXX
+        self.colXY = colXY
+        self.colYY = colYY
+        super().__init__(**kwargs)
+
+    @property
+    def columns(self):
+        return [self.colXX, self.colXY, self.colYY]
+
+    def _func(self, df):
+        return 2*df[self.colXY] / (df[self.colXX] + df[self.colYY])
+
+
+class EffectiveRadius(Functor):
```

Comment:
I'd recommend renaming this class (and `computeReScalar`) to something like `RadiusFromQuadrupole`; it only extracts an effective radius when applied to an ellipse that (like those CModel produces) represents a half-light contour ellipse.  If run on something like SdssShape, it'd produce a second-moment radius instead, for example.